### ISSUE TITLE
MiniWeb crashes after error message

### ISSUE THREAD

**Issue by hayley-jean** [EventStore/EventStore#1297]
Issue raised on google groups [here](https://groups.google.com/forum/#!topic/event-store/xFxlgNnf1wc)

A TCP error appears in the logs, after which the admin UI no longer works.

```[PID:08879:058 2017.04.27 15:24:16.951 INFO  MiniWeb             ] Error while replying from MiniWeb
System.IO.IOException: Unable to write data to the transport connection: The socket has been shut down. ---> System.Net.Sockets.SocketException: The socket has been shut down
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult result) [0x00033] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at System.Net.Sockets.NetworkStream.EndWrite (System.IAsyncResult asyncResult) [0x0005f] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
   --- End of inner exception stack trace ---
  at System.Net.Sockets.NetworkStream.EndWrite (System.IAsyncResult asyncResult) [0x000af] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at System.Net.ResponseStream.EndWrite (System.IAsyncResult ares) [0x00065] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at EventStore.Transport.Http.AsyncQueuedBufferWriter.EndWrite (System.IAsyncResult ar) [0x00000] in <a4bce72d8f6e43208599fe66ae51a7b1>:0 
System.Net.Sockets.SocketException (0x80004005): The socket has been shut down
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult result) [0x00033] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at System.Net.Sockets.NetworkStream.EndWrite (System.IAsyncResult asyncResult) [0x0005f] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
[PID:08879:022 2017.04.27 15:35:50.435 ERROR TcpConnectionMonitor] # <IP1>:20286 1124ms since last Receive started. No completion callback received, but socket status is READY_FOR_RECEIVE
[PID:08879:046 2017.04.27 15:37:52.834 ERROR TcpConnectionMonitor] # <IP3>:20268 1433ms since last Receive started. No completion callback received, but socket status is READY_FOR_RECEIVE
```

**Comment by jrouaix** [comment]
Just happened again with no load on the node, it was idle all the morning : 

```
[PID:29388:019 2017.05.10 02:38:54.248 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: PurgeTimedOutRequests - 74ms. Handler: HttpService.
[PID:29388:019 2017.05.10 02:38:54.248 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: PurgeTimedOutRequests - 74ms. Q: 0/1.
[PID:29388:019 2017.05.10 03:03:00.782 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: PurgeTimedOutRequests - 51ms. Handler: HttpService.
[PID:29388:019 2017.05.10 03:03:00.790 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: PurgeTimedOutRequests - 59ms. Q: 0/2.
[PID:29388:019 2017.05.10 03:19:35.322 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: ReadAllEventsForward - 56ms. Handler: WideningHandler`2.
[PID:29388:019 2017.05.10 03:19:35.324 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: ReadAllEventsForward - 59ms. Q: 1/4.
[PID:29388:023 2017.05.10 03:19:35.325 TRACE InMemoryBus         ] SLOW BUS MSG [bus]: UnwrapEnvelopeMessage - 60ms. Handler: UnwrapEnvelopeHandler.
[PID:29388:023 2017.05.10 03:19:35.325 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [Projection Core #0]: UnwrapEnvelopeMessage - 60ms. Q: 0/4.
[PID:29388:019 2017.05.10 03:34:09.322 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: ReadAllEventsForward - 81ms. Handler: WideningHandler`2.
[PID:29388:019 2017.05.10 03:34:09.329 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: ReadAllEventsForward - 88ms. Q: 2/2.
[PID:29388:025 2017.05.10 03:34:09.322 TRACE InMemoryBus         ] SLOW BUS MSG [bus]: UnwrapEnvelopeMessage - 84ms. Handler: UnwrapEnvelopeHandler.
[PID:29388:025 2017.05.10 03:34:09.330 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [Projection Core #2]: UnwrapEnvelopeMessage - 92ms. Q: 0/4.
[PID:29388:019 2017.05.10 05:08:03.447 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: PurgeTimedOutRequests - 66ms. Handler: HttpService.
[PID:29388:019 2017.05.10 05:08:03.481 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: PurgeTimedOutRequests - 135ms. Q: 0/3.
[PID:29388:023 2017.05.10 05:52:15.959 TRACE InMemoryBus         ] SLOW BUS MSG [bus]: UnwrapEnvelopeMessage - 50ms. Handler: UnwrapEnvelopeHandler.
[PID:29388:023 2017.05.10 05:52:15.959 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [Projection Core #0]: UnwrapEnvelopeMessage - 51ms. Q: 0/4.
[PID:29388:062 2017.05.10 06:07:11.329 INFO  TcpConnection       ] ES TcpConnection closed [06:07:11.329: N<IP4>:20210, L<IP1>:1112, {3980b2b2-33b2-4022-8bf3-580dadd88294}]:Received bytes: 1775665, Sent bytes: 2207109
[PID:29388:062 2017.05.10 06:07:11.414 INFO  TcpConnection       ] ES TcpConnection closed [06:07:11.414: N<IP4>:20210, L<IP1>:1112, {3980b2b2-33b2-4022-8bf3-580dadd88294}]:Send calls: 80598, callbacks: 80598
[PID:29388:062 2017.05.10 06:07:11.414 INFO  TcpConnection       ] ES TcpConnection closed [06:07:11.414: N<IP4>:20210, L<IP1>:1112, {3980b2b2-33b2-4022-8bf3-580dadd88294}]:Receive calls: 80599, callbacks: 80599
[PID:29388:062 2017.05.10 06:07:11.414 INFO  TcpConnection       ] ES TcpConnection closed [06:07:11.414: N<IP4>:20210, L<IP1>:1112, {3980b2b2-33b2-4022-8bf3-580dadd88294}]:Close reason: [Success] Socket closed
[PID:29388:062 2017.05.10 06:07:11.414 INFO  TcpConnectionManager] Connection 'external-normal:<SERVICE_NAME>' [<IP4>:20210, {3980b2b2-33b2-4022-8bf3-580dadd88294}] closed: Success.
[PID:29388:040 2017.05.10 06:07:11.416 DEBUG PersistentSubscripti] Lost connection from <IP4>:20210
[PID:29388:061 2017.05.10 06:07:11.533 INFO  TcpService          ] External TCP connection accepted: [Normal, <IP4>:20216, L<IP1>:1112, {a8c54af9-5357-4cac-bb6a-b247618c52d5}].
[PID:29388:035 2017.05.10 06:07:11.535 INFO  TcpConnectionManager] Connection 'external-normal' ({a8c54af9-5357-4cac-bb6a-b247618c52d5}) identified by client. Client connection name: '<SERVICE_NAME>', Client version: V2.
[PID:29388:019 2017.05.10 07:34:13.846 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: PersistentSubscriptionTimerTick - 72ms. Handler: WideningHandler`2.
[PID:29388:019 2017.05.10 07:34:13.846 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: PersistentSubscriptionTimerTick - 92ms. Q: 1/2.
[PID:29388:019 2017.05.10 07:48:47.685 TRACE InMemoryBus         ] SLOW BUS MSG [MainBus]: PurgeTimedOutRequests - 61ms. Handler: HttpService.
[PID:29388:019 2017.05.10 07:48:47.685 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: PurgeTimedOutRequests - 62ms. Q: 0/1.
[PID:29388:019 2017.05.10 08:19:25.918 TRACE QueuedHandlerAutoRes] SLOW QUEUE MSG [MainQueue]: SendViewChangeProof - 49ms. Q: 0/1.
```

and then this at noon :
```
[PID:29388:061 2017.05.10 12:44:07.250 INFO  MiniWeb             ] Error while replying from MiniWeb
System.IO.IOException: Unable to write data to the transport connection: The socket has been shut down. ---> System.Net.Sockets.SocketException: The socket has been shut down
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult result) [0x00033] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at System.Net.Sockets.NetworkStream.EndWrite (System.IAsyncResult asyncResult) [0x0005f] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
   --- End of inner exception stack trace ---
  at System.Net.Sockets.NetworkStream.EndWrite (System.IAsyncResult asyncResult) [0x000af] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at System.Net.ResponseStream.EndWrite (System.IAsyncResult ares) [0x00065] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at EventStore.Transport.Http.AsyncQueuedBufferWriter.EndWrite (System.IAsyncResult ar) [0x00000] in <a4bce72d8f6e43208599fe66ae51a7b1>:0 
System.Net.Sockets.SocketException (0x80004005): The socket has been shut down
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult result) [0x00033] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
  at System.Net.Sockets.NetworkStream.EndWrite (System.IAsyncResult asyncResult) [0x0005f] in <5071a6e4a4564e19a2eda0f53e42f9bd>:0 
```

**Comment by pgermishuys** [comment]
Closing this due to inactvity. Please would you mind reopening if this is still an issue?
